src/bpfbytecode.cpp

```diff
@@ -83,51 +80,115 @@ BpfProgram &BpfBytecode::getProgramForProbe(const Probe &probe)
       const_cast<const BpfBytecode *>(this)->getProgramForProbe(probe));
 }
 
+namespace {
+/*
+ * Searches the verifier's log for err_pattern. If a match is found, extracts
+ * the name and ID of the problematic helper and throws a HelperVerifierError.
+ *
+ * Example verfier log extract:
+ *     [...]
+ *     36: (b7) r3 = 64                      ; R3_w=64
+ *     37: (85) call bpf_d_path#147
+ *     helper call is not allowed in probe
+ *     [...]
+ *
+ *  In the above log, "bpf_d_path" is the helper's name and "147" is the ID.
+ */
+void maybe_throw_helper_verifier_error(std::string_view log,
+                                       std::string_view err_pattern,
+                                       const std::string &exception_msg_suffix)
```

Comment:
This feels a bit unreliable. Verifier output could change - not sure it is stable.

What is the motivation for this? Doesn't feature probing prevent this?

Reply:
> This feels a bit unreliable. Verifier output could change - not sure it is stable.

This has been around for a long time now, I just moved the code. The truth is that verifier output may change (see #3114 for an example) but we don't have a better way to do this now.

> 
> What is the motivation for this? Doesn't feature probing prevent this?

No, feature probing doesn't help here, some BPF helpers are only allowed in some functions (e.g. `bpf_d_path`, see [here](https://elixir.bootlin.com/linux/latest/source/kernel/trace/bpf_trace.c#L937)).

The motivation is that we want to give the user a nice error message in case he tries to use a helper in an unsupported function so that he doesn't have to deal with the verifier output.